core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPathfinding.kt

```diff
@@ -186,10 +186,20 @@ class STDCMPathfinding(
             queue.add(location)
         }
         val start = Instant.now()
+        var lastFValue = NEGATIVE_INFINITY
         while (true) {
             if (Duration.between(start, Instant.now()).toSeconds() >= pathfindingTimeout)
                 throw OSRDError(ErrorType.PathfindingTimeoutError)
             val endNode = queue.poll() ?: return null
+
+            // Checks that the f-value only goes up, otherwise the heuristic isn't admissible
```

Comment:
A little more context could be nice, something along:
```suggestion
            // Checks that the f-value (best anticipated final value on path) only goes up, otherwise the A* heuristic isn't admissible
```